File: coco/src/main/java/coco/ep/m2/Add.java
package coco.ep.m2;

public interface Add<FT> extends coco.ep.m0.Add<FT>, Exp<FT> {

    default String prettyp() {
        return "(" + convert(getLeft()).prettyp() + "+" + convert(getRight()).prettyp() + ")";
    }
}

File: coco_csharp/CoCo/EP/M2/Add.cs
namespace EP.M2 {
    public interface Add<FT> : Exp<FT>, EP.M0.Add<FT> {

        string Exp<FT>.prettyp() { return "(" + convert(getLeft()).prettyp() + "+" + convert(getRight()).prettyp() + ")"; }
    }
}